(Samples) Transfer in Batch3 (#238)
OfficeDev/office-scripts-docs: docs/resources/samples/add-excel-comments.md

---
title: 'Add comments in Excel'
description: 'Learn how to use Office Scripts to add comments in a worksheet.'
ms.date: 03/18/2021
localization_priority: Normal
---

# Add comments in Excel

> This sample shows how to add comments to a cell with at-mention of a contact.

Reviewer: "At-mention" feels like an inside term.
```suggestion
This sample shows how to add comments to a cell that mention a colleague.
```

Author: @AlexJerabek In support docs, the term "@mention" is used though. Example: [Use @mentions to get someone's attention](https://support.microsoft.com/en-us/office/use-mentions-to-get-someone-s-attention-90701709-5dc1-41c7-aa48-b01d4a46e8c7)

What about...?
```suggestion
This sample shows how to add comments to a cell including @mentioning a colleague.
```

Reviewer: Good call. I like your suggested change.

## Example scenario

* Team lead maintains shift schedule information. Team lead assigns employee ID to shift record.
* Team lead wishes to send email reminder by adding a comment with @mention of the employee by cross referencing the employee email address along with a custom message from the sheet.

> * Employee can then address and respond to the comment.

Reviewer: Small note to look at the grammar of these bullets. I think they could be made into complete sentences.

## Solution

1. The Office Script extracts employee information from the employee sheet.
1. The script then cross references the shift record and adds a comment to the appropriate cell by including the relevant employee email in the team lead's comment.
1. It also removes any existing comment on the cell before adding a new comment.

## Sample Excel file

Download the file <a href="excel-comments.xlsx">excel-comments.xlsx</a> used in this sample and try it out yourself!

## Office Script

> ```ts

Reviewer: Here and elsewhere, let's spell out "TypeScript"
```suggestion
```TypeScript
```
